src/v/kafka/server/handlers/fetch.cc

```diff
@@ -1631,13 +1715,19 @@ fetch_handler::handle(request_context rctx, ss::smp_service_group ssg) {
           }
           octx.response.data.error_code = error_code::none;
           return do_fetch(octx).then([&octx] {
+              auto resp_units_deleter
```

Comment:
Why are these moved here from one part of `octx` to another? I.e., why aren't they just accumulated in the right place in the first place.

Reply:
They're technically being moved from the op_context to the request_context/session resources. And they're not aggregated by shard in the first place due to us needing to free/replace units per ktp throughout the fetch. After the response has been finalized though we don't need the ktp info and can aggregate them by shard for cheaper destruction.